### DISCUSSION TITLE
rdp shows black screen when i connect ubuntu after 3-4 days

### DISCUSSION THREAD

**Discussion by amardeepl11** [neutrinolabs/xrdp#2873] [Q&A]
Hi Team ,

I have configured ubuntu to access from RDP.
I am able to access but after sometime , when i access again , it shows black screen .

to fix the black screen I had to kill the user session from command line, then rdp works fine.

pls note i am using below config.

systemctl status xrdp.service
● xrdp.service - xrdp daemon
     Loaded: loaded (/lib/systemd/system/xrdp.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-12-12 10:09:20 GMT; 16min ago
       Docs: man:xrdp(8)
             man:xrdp.ini(5)
    Process: 131869 ExecStartPre=/bin/sh /usr/share/xrdp/socksetup (code=exited, status=0/SUCCESS)
    Process: 131877 ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS (code=exited, status=0/SUCCESS)
   Main PID: 131878 (xrdp)
      Tasks: 2 (limit: 76625)
     Memory: 11.5M
        CPU: 127ms
     CGroup: /system.slice/xrdp.service
             ├─131878 /usr/sbin/xrdp
             └─131891 /usr/sbin/xrdp

Dec 12 10:09:32 : [INFO ] xrdp_wm_log_msg: sesman connect ok
Dec 12 10:09:32 : [INFO ] sesman connect ok
Dec 12 10:09:32 : [INFO ] sending login info to session manager, please wait...
Dec 12 10:09:33 : [INFO ] xrdp_wm_log_msg: login successful for display 11
Dec 12 10:09:33 : [INFO ] login successful for display 11
Dec 12 10:09:33 : [INFO ] loaded module 'libxup.so' ok, interface size 10296, version 4
Dec 12 10:09:33 : [INFO ] started connecting
Dec 12 10:09:33 : [INFO ] lib_mod_connect: connecting via UNIX socket
Dec 12 10:09:33 : [INFO ] lib_mod_log_peer: xrdp_pid=131891 connected to X11rdp_pid=131895 X11rdp_uid=<Myad uid> X11rdp_gid=<Myad uid> client_ip=::ffff:X.X>
Dec 12 10:09:33 : [INFO ] connected ok

===============================

cat /etc/xrdp/startwm.sh
#!/bin/sh
# xrdp X session start script (c) 2015, 2017 mirabilos \
# published under The MirOS Licence
export DESKTOP_SESSION=ubuntu
export GNOME_SHELL_SESSION_MODE=ubuntu
export XDG_CURRENT_DESKTOP=ubuntu:GNOME

if test -r /etc/profile; then
    . /etc/profile
fi
if test -r /etc/default/locale; then
    . /etc/default/locale

    test -z "${LANG+x}" || export LANG
    test -z "${LANGUAGE+x}" || export LANGUAGE
    test -z "${LC_ADDRESS+x}" || export LC_ADDRESS
    test -z "${LC_ALL+x}" || export LC_ALL
    test -z "${LC_COLLATE+x}" || export LC_COLLATE
    test -z "${LC_CTYPE+x}" || export LC_CTYPE
    test -z "${LC_IDENTIFICATION+x}" || export LC_IDENTIFICATION
    test -z "${LC_MEASUREMENT+x}" || export LC_MEASUREMENT
    test -z "${LC_MESSAGES+x}" || export LC_MESSAGES
    test -z "${LC_MONETARY+x}" || export LC_MONETARY
    test -z "${LC_NAME+x}" || export LC_NAME
    test -z "${LC_NUMERIC+x}" || export LC_NUMERIC
    test -z "${LC_PAPER+x}" || export LC_PAPER
    test -z "${LC_TELEPHONE+x}" || export LC_TELEPHONE
    test -z "${LC_TIME+x}" || export LC_TIME
    test -z "${LOCPATH+x}" || export LOCPATH
fi

if test -r /etc/profile; then
    . /etc/profile
fi
test -x /etc/X11/Xsession && exec /etc/X11/Xsession
exec /bin/sh /etc/X11/Xsession

===========

![image](https://github.com/neutrinolabs/xrdp/assets/139690905/d0ccebee-7692-4c89-a0c3-8c556676a928)

**Comment by matt335672** [comment]
There's likely to be some problem with your session if you're leaving it for so long. While you're not connected, the RDP server is doing nothing at all. For example, you may have a web session open which is using all of your available memory.

You can clearly get access to the machine when the black screen occurs. Next time it happens, check you've not run out of memory, then kill applications one-by-one until the session recovers. That will give us a clue.

**Reply by amardeepl11** [reply]
Hi @matt335672 ,

Thanks for confirmation .

can you suggest on this .

i start connecting my ubuntu via rdp , its connects fine on day1

then i disconnect my rdp session from windows and go back to home
i come back next morning in office and try to connect via rdp , it shows blank screen and does not connect until i go back in server via ssh and run this command - sudo pkill -KILL -u <username>.

can you please help me how can i connect without killing my user session.
Thanks.

**Reply by matt335672** [reply]
@amardeepl11 - a couple of things:-
1) Check the system log for messages from the Linux OOM killer relating to getting low on memory.
2) If it's not memory, kill your applications on-by-one from the command line until you find the one which is blocking the session.
